### ISSUE TITLE
Documentation for `Binomial` is potentially misleading

### ISSUE THREAD

**Issue by fingolfin** [gap-system/gap#957]
It says
"Binomial( n, k ) returns the binomial coefficient {n choose k} of integers n and k, which is defined as n! / (k! (n-k)!) (see Factorial (16.1-1)). We define {0 choose 0} = 1, {n choose k} = 0 if k < 0 or n < k, and {n choose k} = (-1)^k {-n+k-1 choose k} if n < 0, which is consistent with the equivalent definition {n choose k} = {n-1 choose k} + {n-1 choose k-1}."

Now, what is `Binomial(-1,1)`? Well, $n=-1 < k = 1$, so it should be 0. But also $n<0$, so it should be `-Binomial(1, 1) = -1`.

It actually is -1, and I think that's fine. But I think we should clarify this. Perhaps as simple a change as replacing `or n < k` by `or 0 <= n < k` would be sufficient?

**Comment by ChrisJefferson** [comment]
I agree it should be made consistent, and I don't find either answer particularly better than the other -- I might have just said no negative numbers myself, but that ship has long-since sailed.

Fixing the docs seems best, just in case someone is relying on this behaviour in their code.

**Comment by hulpke** [comment]
(As I'm currently teaching combinatorics)
The most common generalization of binomial coefficients ${n\choose k}$ are as power series coefficients in the expansion of $(1+t)^n$ for arbitrary $n$. (see e.g. https://en.wikipedia.org/wiki/Binomial_coefficient#Generalization_and_connection_to_the_binomial_series )
In this context -1 would be the appropriate answer.

**Comment by frankluebeck** [comment]
The current desciption is indeed bad. How about the following:

```
 16.1-2 Binomial
  
  ‣ Binomial( n, k ) ──────────────────────────────────────────────── function
  
  returns  the  binomial coefficient {n choose k} of integers n and k. This is
  defined  by  the conditions {n choose k} = 0 for k < 0, {0 choose 0} = 1 and
  the  relation {n choose k} = {n-1 choose k} + {n-1 choose k-1} for all n and
  k.  There  are  many ways of describing this function. For example, if n ≥ 0
  and  0 ≤ k ≤ n, then {n choose k} = n! / (k! (n-k)!) and for n < 0 and k ≥ 0
  we have {n choose k} = (-1)^k {-n+k-1 choose k}.
  
  {n choose k} is the number of combinations with k elements, i.e., the number
  of  subsets  with  k elements, of a set with n elements. {n choose k} is the
  coefficient  of  the  term  x^k  of  the  polynomial (x + 1)^n, which is the
  generating function for {n choose .}, hence the name.
```

If you are happy with this description I can provide a pull request.

**Comment by fingolfin** [comment]
First off, I absolutely think that -1 is the "correct" result, so I am fine with that. But I had a student who was thoroughly confused by the documentation; luckily she's bright and resolved the matter in the end by finding the source code for `Binomial` and settling it that way... :-)

@frankluebeck Perhaps start the second paragraph with "For positive n, ..." or so? Otherwise, I think this is fine.

**Comment by fingolfin** [comment]
Resolved by #967
